mmv1/products/networkconnectivity/Spoke.yaml

```diff
@@ -203,11 +209,13 @@ properties:
               type: String
               description: The URI of the virtual machine resource
               immutable: true
+              required: true
               diff_suppress_func: 'tpgresource.CompareSelfLinkOrResourceName'
             - name: 'ipAddress'
               type: String
               description: The IP address on the VM to use for peering.
               immutable: true
+              required: true
```

Comment:
Are these two fields required in the API Create request? I wonder if it is a breaking change for the users to mark the fields required.

Reply:
Yes these fields are currently required in the API

```
gcloud network-connectivity spokes linked-router-appliances create spoke-lan-vyos-transit-1ncc-usw1 --region us-west1 \
        --router-appliance=instance="https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01"
ERROR: (gcloud.network-connectivity.spokes.linked-router-appliances.create) argument --router-appliance: Key [ip] required in dict arg but not provided
Usage: gcloud network-connectivity spokes linked-router-appliances create (SPOKE : --region=REGION) --hub=HUB --router-appliance=[instance=INSTANCE],[ip=IP] [optional flags]
  optional flags may be  --async | --description | --group | --help |
                         --include-import-ranges | --labels | --region |
                         --site-to-site-data-transfer

For detailed information on this command and its flags, run:
  gcloud network-connectivity spokes linked-router-appliances create --help
```

Reply:
That is correct, this field is currently required in the API

https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/Shared.Types/Spoke#routerapplianceinstance

```
 gcloud network-connectivity spokes linked-router-appliances create spoke-lan-vyos-transit-1ncc-usw1 --region us-west1 \
        --hub="projects/rteller-demo-hst-e265-aaaa/locations/global/hubs/hub-lan-1ncc-gbl0" \
        --router-appliance=instance="https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01"
ERROR: (gcloud.network-connectivity.spokes.linked-router-appliances.create) argument --router-appliance: Key [ip] required in dict arg but not provided
Usage: gcloud network-connectivity spokes linked-router-appliances create (SPOKE : --region=REGION) --hub=HUB --router-appliance=[instance=INSTANCE],[ip=IP] [optional flags]
  optional flags may be  --async | --description | --group | --help |
                         --include-import-ranges | --labels | --region |
                         --site-to-site-data-transfer

For detailed information on this command and its flags, run:
  gcloud network-connectivity spokes linked-router-appliances create --help
  
  ```